src/hooks/analytics/useTrackAppOpened.ts

```diff
@@ -1,26 +1,25 @@
-import { usePostHog } from 'posthog-react-native';
 import React from 'react';
 import { AppState, NativeEventSubscription } from 'react-native';
-
-const APPLICATION_OPENED_EVENT = 'Application Opened';
+import useTrackEvent from 'hooks/analytics/useTrackEvent';
+import { Events } from 'types/analytics';
 
 /**
  * Hook to track when the user opens the applications.
  */
 const useTrackAppOpened = () => {
+  const trackEvent = useTrackEvent();
   const [appStarted, setAppStarted] = React.useState(false);
-  const postHog = usePostHog();
 
   React.useEffect(() => {
     let subscription: NativeEventSubscription | undefined;
-    if (!appStarted && postHog) {
+    if (!appStarted) {
```

Comment:
Here in the situation in wich `postHog` is `undefined` the `trackEvent` will return without capturing the   `ApplicationOpened` event but the `appStarted` will be set to `true` causing the application to not correclty capture the `ApplicationOpened` event.
I suggest to modify `trackEvent` to make it return a boolean value that tells if the event has been captured or not so that we can correctly update the `appStarted` value.